hw/ip/i2c/doc/_index.md

```diff
@@ -14,7 +14,7 @@ See that document for integration overview within the broader top level system.
 ## Features 
 
 - Two-pin clock-data parallel bidirectional external interface 
-- The initial revision only supports I2C as a Host ("I2C Master"<sup>1</sup>).
+- Support for I2C Host ("I2C Master"<sup>1</sup>) and I2C Target ("I2C Slave"<sup>1</sup>) device modes
```

Comment:
Do we need to clarify how the Host/Target mode is configured, either at compiler time (via an RTL parameter) or runtime (via a configuration register)? Since I2C will be instantiated multiple times thus the runtime configuration incurs area and power overhead.

Reply:
Good point. Considering past discussions of I2C combo IP area which occurred prior to this PR, we are leaning towards runtime configuration via the registers ENABLEHOST/ENABLETARGET. Should we state this here explicitly?

Reply:
Instead, I think we should both RTL parameter (compile-time), for instance, "parameter I2C_CONFIG HOST/TARGET/COMBO". Based on the value of I2C_CONFIG, the RTL can be added/eliminated via #ifdef for the area and power reduction whenever the IP is instantiated. 
Specific to combo RTL (I2C_CONFIG = combo), we can have a register bit, named MODE, for runtime configuration. This bit is set by SW (ENABLEHOST/ENABLETARGET) depending on the function supported by HW. For instance, if the RTL only supports TARGET mode, SW takes responsibility to program MODE bit to ENABLETARGET otherwise, RTL would not function correctly.
It should be noted, if the logic and register interface for HOST and TARGET in RTL are separate, the MODE bit can be eliminated

Reply:
I thought we had concluded that we were going to use the combo I2C module? Which can do both host and target at the same time (hence the need for four fifos).

If we do make this a parameter we need a way for the sw to tell which version is instantiated otherwise it is going to get much more complicated when we have multiple different chips but want a generic software base.

BTW if we think the block is too big, we could consider making the fifos smaller (the diagram says they are 128 entry today and I suspect 64 or 32 would be ok for most uses).